Contrib/Fastcluster/fastcluster.py

```diff
@@ -0,0 +1,61 @@
+"""
+This script performs fast clustering of SMILES
+
+Clustering method is repeated bi section, the method looks like -k-means.
+To use this script, the user needs to install bayon at first.
+Input format: Tab separated SMILES strings (SMILES \t molID \n ...)
+Please see more details in README.
+
+"""
+
+import argparse
+import subprocess
+import pickle
+import os
+from rdkit import Chem
+from rdkit.Chem import AllChem
+
+def getArgParser():
+    """ Create the argument parser """
+    parser = argparse.ArgumentParser("Fast clustering for chemoinformatics")
+    parser.add_argument("input", help="filename of input file")
+    parser.add_argument("nclusters", metavar="N", help="the number of clusters")
+    parser.add_argument("--output",  help="filename of output, tab separated format", default="clustered.tsv")
+    parser.add_argument("--centroid", metavar="CENTROID",  help="filename of centroid information. tab separated format", default="centroid.tsv")
+    return parser
+
+def smi2fp(molid, smiles):
+    mol = Chem.MolFromSmiles(smiles)
+    onbits = AllChem.GetMorganFingerprintAsBitVect(mol, 2).GetOnBits()
+    row = molid
+    for bit in onbits:
+        row += "FP_{}\t1.0".format(bit)
+    row += "\n" 
+    return row 
+
+
+if __name__ == "__main__":
+    parser = getArgParser()
+    args = parser.parse_args()
+    with open(args.input, "r") as inputf:
+        with open("fp.tsv", "w") as tempf:
```

Comment:
If you decide to only support Python 3, these two lines could also be written as:
```
 with open(args.input, "r") as inputf, open("fp.tsv", "w") as tempf:
```